### PR TITLE
Change os functions to new arg style

### PR DESCRIPTION
**Pull request by palaviv**

### CODE REVIEW

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -83,17 +83,11 @@ fn make_path(_vm: &VirtualMachine, path: PyStringRef, dir_fd: &DirFd) -> PyStrin
     }
 }
 
-pub fn os_close(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(fileno, Some(vm.ctx.int_type()))]);
-
-    let raw_fileno = objint::get_value(&fileno);
-
```
Comment (coolreader18) [inline vm/src/stdlib/os.rs]: This could be an `i64` now

File: vm/src/stdlib/os.rs [hunk 2]
```diff
@@ -83,17 +83,11 @@ fn make_path(_vm: &VirtualMachine, path: PyStringRef, dir_fd: &DirFd) -> PyStrin
...
+fn os_close(fileno: PyIntRef, _vm: &VirtualMachine) {
     //The File type automatically closes when it goes out of scope.
     //To enable us to close these file descriptors (and hence prevent leaks)
     //we seek to create the relevant File and simply let it pass out of scope!
-    rust_file(raw_fileno.to_i64().unwrap());
-
-    Ok(vm.get_none())
```
Comment (coolreader18) [inline vm/src/stdlib/os.rs]: I know that this was here before the change, but using `drop(file)` is more explicit as to what's happening.

Comment (palaviv) [reply vm/src/stdlib/os.rs]: Better :)

Comment (palaviv) [reply vm/src/stdlib/os.rs]: I think that the drop don't work as expected on windows. I will try to see what happens when I revert it.

**Pull request merged**